### PR TITLE
refactor: `main.py`

### PR DESCRIPTION
**Pull request by Aleksander-Karlsson**
## Description
*Include a description of the change done*
Clean up main.py:
- `FileWriter` - replaced with single context managed write, reducing IO operations.
- `ProgressStatus` - replaced with imported tqdm package.
- Simplify create method somewhat.

# Why
*The motivation for the change (if applicable)*
Reduce technical depth.

# How
*How the change is implemented (if it's not self evident in the diff)*

Close: #XXX

# Checklist:
Before submitting this PR, please make sure:

- [x] I am complying with the [contributing](https://equinor.github.io/completor/contribution_guide) doc
- [x] My code builds clean without any errors or warnings
- [x] I have added/extended tests that prove my fix is effective or that my feature works
- [x] I have made corresponding changes to the documentation, and it builds correctly

### CODE REVIEW

**Review by cutn94 — APPROVED** [review]

File: completor/parse.py [hunk 1]
```diff
@@ -245,6 +245,8 @@ def read_schedule_keywords(
     content = deepcopy(content)
     used_index = np.asarray([-1])
     collections = []
+    if optional_keywords is None:
+        optional_keywords = []
```
Comment (Aleksander-Karlsson) [inline completor/parse.py:249]:
```suggestion
    optional_keywords = [] if optional_keyword is None else optional_keywords
```

File: completor/parse.py [hunk 2]
```diff
@@ -245,6 +245,7 @@ def read_schedule_keywords(
     content = deepcopy(content)
     used_index = np.asarray([-1])
     collections = []
+    optional_keywords = [] if optional_keyword is None else optional_keywords
```
Comment (Aleksander-Karlsson) [inline completor/parse.py:248]:
```suggestion
    optional_keywords = [] if optional_keywords is None else optional_keywords
```

File: completor/create_output.py [hunk 3]
```diff
@@ -363,12 +360,12 @@ def make_wsegicv(self, lateral: int) -> None:
     def make_wsegdar(self) -> None:
         """Print WSEGDAR to file."""
         if self.df_wsegdar.shape[0] > 0:
-            self.print_wsegdar += po.print_wsegdar(self.df_wsegdar, self.iwell + 1) + "\n"
+            self.print_wsegdar += po.print_wsegdar(self.df_wsegdar, self.well_number + 1) + "\n"
```
Comment (MegaAnda123) [inline completor/create_output.py:363]: Using `"".join` on a list of strings is way faster.'
https://stackoverflow.com/questions/2414667/python-string-class-like-stringbuilder-in-c

**Review by MegaAnda123 — APPROVED** [review]
One comment

**Pull request merged**